fix(transformer/class-properties): don't transform properties that have `declare` modifier

Dunqing opened the pull request:
* Fixes #13733

Fixes the issue where `declare` fields were incorrectly transformed instead of being removed, causing inheritance issues where child `declare` fields would interfere with parent class property initialization.

Only in the `enter_*` phase do we have to skip the `declare` field; in the exit phase, the `declare` fields have already been removed in the `TypeScript` plugin.

overlookmotel reviewed (approved):
Pushed a couple of trivial commits with changes to comments. Hope that's OK.

Please see comment below. If you prefer to address that in follow-up, feel free to merge this PR as is, because I think the code is redundant anyway.

I just realised that we don't debug assert that `classes_stack` is emptied in `exit_program` as we should. I think that'd be a good idea, to make sure the stack doesn't get out of sync due to skipping `push` or `pop` due to `declare == true`.

overlookmotel commented on crates/oxc_transformer/src/es2022/class_properties/class.rs:
```diff
@@ -402,9 +407,6 @@ impl<'a> ClassProperties<'a, '_> {
     ) {
         // Ignore TS class declarations
         // TODO: Is this correct?
-        if class.declare {
-            return;
-        }
```
Yes, sounds good.

Dunqing replied:
Reverted, I am merging it.

Dunqing committed:
fix(transformer/class-properties): don't transform properties that have `declare` modifier (#13766)

* Fixes #13733

Fixes the issue where `declare` fields were incorrectly transformed instead of being removed, causing inheritance issues where child `declare` fields would interfere with parent class property initialization.

Only in the `enter_*` phase do we have to skip the `declare` field; in the exit phase, the `declare` fields have already been removed in the `TypeScript` plugin.
